### PR TITLE
Frontend component

### PR DESCRIPTION
**Pull request by tomasedl**
java script part should be done but need some cleaning procedure. If you find anything, please ping me on discord

### CODE REVIEW

File: formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java [hunk 1]
```diff
@@ -45,4 +59,22 @@
     }
 
     //todo: engineer can only view all, view one, add or edit... but cannot delete
+    @ApiOperation(value = "Create new component")
+    @PutMapping(value = ApiUris.ROOT_URI_COMPONENTS, consumes = MediaType.APPLICATION_JSON_VALUE)
+    @RolesAllowed(Role.ROLE_ENGINEER)
```
Comment (Jakub-Fajkus) [inline formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java:64]: Tohle by mel byt schopny delat asi i manager :smile:

File: formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java [hunk 2]
```diff
@@ -45,4 +59,22 @@
...
+    public RestResponse<ComponentAPIDTO> createComponent(@RequestBody ComponentAPIDTO component) {
+        Long id = componentFacade.createComponent(new ComponentDTO(component.getType(), component.getName(), null));
+        return new RestResponse<>(dozer.mapTo(componentFacade.getComponentById(id), ComponentAPIDTO.class), ResponseStatuses.OK);
+    }
+
+    @ApiOperation(value = "Update information about given component")
+    @PatchMapping(value = ApiUris.ROOT_URI_COMPONENT, produces = MediaType.APPLICATION_JSON_VALUE)
+    @RolesAllowed(Role.ROLE_ENGINEER)
```
Comment (Jakub-Fajkus) [inline formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java:72]: Tohle by mel byt schopny delat asi i manager :smile:

File: formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java [hunk 3]
```diff
@@ -45,4 +59,22 @@
     }
 
     //todo: engineer can only view all, view one, add or edit... but cannot delete
```
Comment (Jakub-Fajkus) [inline formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java:61]: todo muzem uz asi smazat

File: formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java [hunk 4]
```diff
@@ -45,4 +59,22 @@
...
+    @ApiOperation(value = "Create new component")
+    @PutMapping(value = ApiUris.ROOT_URI_COMPONENTS, consumes = MediaType.APPLICATION_JSON_VALUE)
```
Comment (Jakub-Fajkus) [inline formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java:63]: tady by asi melo byt i `produces = json`

File: formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java [hunk 5]
```diff
@@ -45,4 +59,22 @@
...
+    @RolesAllowed(Role.ROLE_ENGINEER)
+    public RestResponse<ComponentAPIDTO> createComponent(@RequestBody ComponentAPIDTO component) {
+        Long id = componentFacade.createComponent(new ComponentDTO(component.getType(), component.getName(), null));
+        return new RestResponse<>(dozer.mapTo(componentFacade.getComponentById(id), ComponentAPIDTO.class), ResponseStatuses.OK);
```
Comment (Jakub-Fajkus) [inline formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java:67]: muzes rovnou zavolat metodu `getComponent`

File: formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java [hunk 6]
```diff
@@ -45,4 +59,22 @@
...
+    }
+
+    @ApiOperation(value = "Update information about given component")
+    @PatchMapping(value = ApiUris.ROOT_URI_COMPONENT, produces = MediaType.APPLICATION_JSON_VALUE)
+    @RolesAllowed(Role.ROLE_ENGINEER)
+    public RestResponse<ComponentAPIDTO> updateComponent(@ApiParam(value = "Id of a component for edit") @PathVariable Long id, @RequestBody UpdateComponentAPIDTO component) {
+        CarDTO carDTO = component.getCar() == null ? null : carFacade.getCarById(component.getCar());
```
Comment (Jakub-Fajkus) [inline formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java:74]: Proc tady potrebujeme auto? bez tohoto by se te komponente auto `odnastavilo` :smile: ?

File: formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/security/AuthTokenFilter.java [hunk 7]
```diff
@@ -36,7 +36,7 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
             String jwt = parseJwt(request);
             if (true || jwt != null && jwtUtils.validateJwtToken(jwt)) {
 //                String username = jwtUtils.getUserNameFromJwtToken(jwt);
-                String username = "manager"; //todo: :D temporary
+                String username = "engineer1"; //todo: :D temporary//ENGINEER engineer1
```
Comment (Jakub-Fajkus) [inline formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/security/AuthTokenFilter.java:39]: tohle bych tu radeji nedaval :smile:

File: formula1-rest/src/main/webapp/public/components/componentComponentAdd.js [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+            }, {
+                headers: {
+                    "Content-type": "application/json",
+                    "Authorization": `Bearer ${store.$jwt}`,
+                }
+            })
+                .then(response => {
+                    console.log("Success:", response);
+
+                    this.name = response.data.data.name;
```
Comment (Jakub-Fajkus) [inline formula1-rest/src/main/webapp/public/components/componentComponentAdd.js:63]: tady je to asi celkove zbytecne ukladat do this.name... hned potom "presmerovavas" na jinou komponentu

File: formula1-rest/src/main/webapp/public/components/componentComponentAdd.js [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+                    this.$emit('add-component');
```
Comment (Jakub-Fajkus) [inline formula1-rest/src/main/webapp/public/components/componentComponentAdd.js:65]: tady mi nejak po vytvoreni komponenty nefunguje "presmerovani"

File: formula1-rest/src/main/webapp/public/components/componentComponentDetail.js [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+                    "Content-type": "application/json",
+                    "Authorization": `Bearer ${store.$jwt}`,
+                }
+            })
+                .then(response => {
+                    console.log("Success:", response);
+
+                    this.name = response.data.data.name;
+
+                    functions.showSuccessNotification("Component successfully edited, maybe :D")
```
Comment (Jakub-Fajkus) [inline formula1-rest/src/main/webapp/public/components/componentComponentDetail.js:82]: maybe :trollface:

**Review by Jakub-Fajkus — CHANGES_REQUESTED** [review]

File: formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java (hunk 6 above)
Comment (tomasedl) [reply formula1-rest/src/main/java/cz/muni/pa165/teamwhite/formula1/rest/controller/ComponentController.java:74]: doreseno

**Review by Jakub-Fajkus — APPROVED** [review]

**Pull request merged**